src/contentnormalizer.js

```diff
@@ -0,0 +1,157 @@
+ * returns `true` or `false`. Based on this result normalizer applies filters to given data.
+ *
+ * {@link module:paste-from-office/contentnormalizer~FilterFunction Filters} are function, which are run sequentially, as they were added.
+ * Each filter gets data transformed by the previous one.
+ *
+ * Example definition:
+ *
+ * 	const normalizer = new ContentNormalizer( contentHtml =>
+ * 		contentHtml.includes( 'docs-internal-guid' )
+ * 	);
+ *
+ * 	normalizer.addFilter( ( { data, documentFragment, writer } ) => {
+ * 		// filter's content, which transforms data in clipboard
+ * 	} );
+ *
+ * Normalizers are stored inside Paste from Office plugin and are run on
+ * {@link module:clipboard/clipboard~Clipboard#event:inputTransformation inputTransformation event}. Below example is simplified and show
+ * how to call normalizer directly on clipboard event, what has happen inside
+ * {@link module:paste-from-office/pastefromoffice~PasteFromOffice} plugin.
+ *
+ * 	editor.plugins.get( 'Clipboard' ).on( 'inputTransformation', ( evt, data ) => {
+ * 		normalizer.transform( data );
+ * 	} );
+ *
+ * @class
+ */
+export default class ContentNormalizer {
+	/**
+	 * Initialize Content Normalizer.
+	 *
+	 * @param {Function} activationTrigger The function which checks for what content should be applied this normalizer.
+	 * It takes an HTML string from the `text/html` dataTransfer as an argument and have to return a boolean value
+	 */
+	constructor( activationTrigger ) {
+		/**
+		 * Keeps a reference to the activation trigger function. The function is used to check if current Content Normalizer instance
+		 * should be applied for given input data. Check is made during the {@link #transform}.
+		 *
+		 * @private
+		 * @type {Function}
+		 */
+		this._activationTrigger = activationTrigger;
+
+		/**
+		 * Keeps a reference to registered filters with {@link #addFilter} method.
+		 *
+		 * @private
+		 * @type {Set}
+		 */
+		this._filters = new Set();
+	}
+
+	/**
+	 * Method checks if passed data should have applied {@link #_filters} registerd in this Content Normalizer.
+	 * If yes, then data are transformed and marked with a flag `isTransformedWithPasteFromOffice=true`.
+	 * In other case data are not modified.
+	 *
+	 * Please notice that presence of `isTransformedWithPasteFromOffice` flag in input data prevent transformation.
+	 * This forbid of running the same normalizer twice or running multiple normalizers over the same data.
+	 *
+	 * @param data input data object it should preserve structure defined in
+	 * {@link module:clipboard/clipboard~Clipboard#event:inputTransformation Clipboard#inputTransformation event}.
+	 */
+	transform( data ) {
+		const html = data.dataTransfer && data.dataTransfer.getData( 'text/html' );
+		const dataReadFirstTime = data.isTransformedWithPasteFromOffice === undefined;
+		const hasHtmlData = !!html;
+
+		if ( hasHtmlData && dataReadFirstTime && this._activationTrigger( html ) ) {
```

Comment:
The when and how many times to run the content normalizer should be the responsibility of the PFO plugin ATM. Let's make Normalizer less clever ;)